platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java

```diff
@@ -304,10 +302,21 @@ public GossipEvent getBaseEvent() {
     }
 
     /**
-     * @return The hashed part of a base event
+     * Check if the event has a self parent.
+     *
+     * @return true if the event has a self parent
+     */
+    public boolean hasSelfParent() {
+        return baseEvent.getSelfParent() != null;
+    }
+
+    /**
+     * Check if the event has other parents.
+     *
+     * @return true if the event has other parents
      */
-    public BaseEventHashedData getHashedData() {
-        return baseEvent.getHashedData();
+    public boolean hasOtherParent() {
```

Comment:
```suggestion
    public boolean hasOtherParents() {
```

Reply:
This is an issue with the fact that we only half-way support multiple other parents. We added support in some parts of the code, but not in others. Consensus, which uses this method, does not support multiple other parents. So I think it would be confusing if only this method implied there could be multiple parents, but not others.